test/functional/lua/utility_functions_spec.lua

```diff
@@ -106,3 +108,96 @@ describe('vim.stricmp', function()
     eq(1, funcs.luaeval('vim.stricmp("\\0C\\0", "\\0B\\0")'))
   end)
 end)
+
+describe("vim.split", function()
+  it("works", function()
+    local split = function(str, sep)
+      return funcs.luaeval('vim.split(_A.s, _A.e)', { s = str, e = sep })
+    end
+
+    local tests = {
+      { "a,b", ",", false, { 'a', 'b' } },
+      { ":aa::bb:", ":", false, { '', 'aa', '', 'bb', '' } },
+      { "::ee::ff:", ":", false, { '', '', 'ee', '', 'ff', '' } },
+      { "ab", ".", false, { '', '', '' } },
+      { "a1b2c", "[0-9]", false, { 'a', 'b', 'c' } },
+      { "xy", "", false, { 'x', 'y' } },
+      { "here be dragons", " ", false, { "here", "be", "dragons"} },
+      { "axaby", "ab?", false, { '', 'x', 'y' } },
+      { "f v2v v3v w2w ", "([vw])2%1", false, { 'f ', ' v3v ', ' ' } },
+      { "x*yz*oo*l", "*", true, { 'x', 'yz', 'oo', 'l' } },
+    }
+
+    for _, t in ipairs(tests) do
+      eq(t[4], split(t[1], t[2], t[3]))
+    end
+
+    local loops = {
+      { "abc", ".-", { 'a', 'b', 'c'} },
+    }
+
+    for _, t in ipairs(loops) do
+      local status, err = pcall(split, t[1], t[2])
+      eq(false, status)
+      assert(string.match(err, "Infinite loop detected"))
+    end
+  end)
+end)
+
+describe("vim.inspect", function()
+  it('works', function()
+    command("source runtime/plugin/lua.vim") 
+    -- just make sure it basically works, it has its own test suite
+    local inspect = function(t, opts)
+      return funcs.luaeval('vim.inspect(_A.t, _A.o)', { t = t, o = opts })
+    end
+
+    eq('2', inspect(2))
+
+    local i = inspect({ a = { b = 1 } }, { newline = '+', indent = '' })
+    eq('{+a = {+b = 1+}+}', i)
+  end)
+end)
+
+describe("vim.trim", function()
+  it('works', function()
+    local trim = function(s)
+      return funcs.luaeval('vim.trim(_A.s)', { s = s })
+    end
+
+    local trims = {
+      { "   a", "a" },
+      { " b  ", "b" },
+      { "\tc" , "c" },
+      { "r\n", "r" },
+    }
+
+    for _, t in ipairs(trims) do
+      assert(t[2], trim(t[1]))
+    end
+
+    local status, err = pcall(trim, 2)
+    eq(false, status)
+    assert(string.match(err, "Only strings can be trimmed"))
+  end)
+end)
+
+describe("vim.deepcopy", function()
+  it("works", function()
+    local deepcopy = function(t)
+      return funcs.luaeval('vim.deepcopy(_A.t)', { t = t })
+    end
+
+    local a = { x = { 1, 2 }, y = 5}
+    local b = deepcopy(a)
+
+    local count = 0
+    for _ in pairs(b) do count = count + 1 end
+
+    eq(1, b.x[1])
+    eq(2, b.x[2])
+    eq(5, b.y)
+    eq(2, count)
+    neq(tostring(a), tostring(b))
```

Comment:
This unfortunately doesn't check that a deepcopy was done, calling the identity function (in nvim process) will give the same result.

Reply:
Not sure I catch you, afaik `tostring` on a table value gives its pointer address, so if it's different, it's not the same table. Why would you get that for the identity function?

Reply:
Because the msgpck RPC back and forth will force to allocate a new table anyway.

Reply:
Ahhh sure. I'll fix that, thanks :)

Reply:
Ok, fixed it, I also dutifully confirmed that the test fails when just using `local b = a`.